2025-07-13 10:35:28 [food.spiders.ingredients] WARNING: Crawling completed in 0:01:40.521060
2025-07-13 16:16:34 [food.spiders.ingredients] WARNING: Crawling completed in 0:01:29.790918
2025-07-13 16:17:58 [food.spiders.ingredients] WARNING: Crawling completed in 0:00:58.012415
2025-07-13 16:18:48 [food.spiders.ingredients] WARNING: Crawling completed in 0:00:46.402670
2025-07-14 09:57:56 [food.spiders.ingredients] WARNING: Crawling completed in 0:01:06.280529
2025-07-14 10:06:31 [scrapy.core.engine] ERROR: Scraper close failure
Traceback (most recent call last):
  File "C:\Users\hp\bb-team-info-by-ingredient\.venv\Lib\site-packages\twisted\internet\defer.py", line 1088, in _runCallbacks
    current.result = callback(  # type: ignore[misc]
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\hp\bb-team-info-by-ingredient\food\food\pipelines.py", line 184, in close_spider
    self.wb.save("foods.xlsx")
  File "C:\Users\hp\bb-team-info-by-ingredient\.venv\Lib\site-packages\openpyxl\workbook\workbook.py", line 386, in save
    save_workbook(self, filename)
  File "C:\Users\hp\bb-team-info-by-ingredient\.venv\Lib\site-packages\openpyxl\writer\excel.py", line 291, in save_workbook
    archive = ZipFile(filename, 'w', ZIP_DEFLATED, allowZip64=True)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\hp\AppData\Local\Programs\Python\Python311\Lib\zipfile.py", line 1294, in __init__
    self.fp = io.open(file, filemode)
              ^^^^^^^^^^^^^^^^^^^^^^^
PermissionError: [Errno 13] Permission denied: 'foods.xlsx'
2025-07-14 10:06:31 [food.spiders.ingredients] WARNING: Crawling completed in 0:01:03.201723
2025-07-14 10:10:43 [food.spiders.ingredients] WARNING: Crawling completed in 0:01:04.890334
2025-07-14 10:45:15 [food.spiders.ingredients] WARNING: Crawling completed in 0:01:02.501642
2025-07-14 11:02:48 [food.spiders.ingredients] WARNING: Crawling completed in 0:01:12.785559
2025-07-14 11:07:54 [scrapy.core.engine] ERROR: Scraper close failure
Traceback (most recent call last):
  File "C:\Users\hp\bb-team-info-by-ingredient\.venv\Lib\site-packages\twisted\internet\defer.py", line 1088, in _runCallbacks
    current.result = callback(  # type: ignore[misc]
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\hp\bb-team-info-by-ingredient\food\food\pipelines.py", line 197, in close_spider
    self.index_to_column_letter(start_column),
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: XSLXPipeline.index_to_column_letter() takes 1 positional argument but 2 were given
2025-07-14 11:07:54 [food.spiders.ingredients] WARNING: Crawling completed in 0:01:05.495909
2025-07-14 11:13:15 [scrapy.core.engine] ERROR: Scraper close failure
Traceback (most recent call last):
  File "C:\Users\hp\bb-team-info-by-ingredient\.venv\Lib\site-packages\twisted\internet\defer.py", line 1088, in _runCallbacks
    current.result = callback(  # type: ignore[misc]
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\hp\bb-team-info-by-ingredient\food\food\pipelines.py", line 198, in close_spider
    self.index_to_column_letter(start_column),
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: XSLXPipeline.index_to_column_letter() takes 1 positional argument but 2 were given
2025-07-14 11:13:15 [food.spiders.ingredients] WARNING: Crawling completed in 0:01:06.066959
2025-07-14 11:22:23 [twisted] CRITICAL: Unhandled error in Deferred:
2025-07-14 11:22:23 [twisted] CRITICAL: 
Traceback (most recent call last):
  File "C:\Users\hp\bb-team-info-by-ingredient\.venv\Lib\site-packages\twisted\internet\defer.py", line 2017, in _inlineCallbacks
    result = context.run(gen.send, result)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\hp\bb-team-info-by-ingredient\.venv\Lib\site-packages\scrapy\crawler.py", line 156, in crawl
    self.engine = self._create_engine()
                  ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\hp\bb-team-info-by-ingredient\.venv\Lib\site-packages\scrapy\crawler.py", line 169, in _create_engine
    return ExecutionEngine(self, lambda _: self.stop())
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\hp\bb-team-info-by-ingredient\.venv\Lib\site-packages\scrapy\core\engine.py", line 111, in __init__
    self.scraper: Scraper = Scraper(crawler)
                            ^^^^^^^^^^^^^^^^
  File "C:\Users\hp\bb-team-info-by-ingredient\.venv\Lib\site-packages\scrapy\core\scraper.py", line 107, in __init__
    self.itemproc: ItemPipelineManager = itemproc_cls.from_crawler(crawler)
                                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\hp\bb-team-info-by-ingredient\.venv\Lib\site-packages\scrapy\middleware.py", line 77, in from_crawler
    return cls._from_settings(crawler.settings, crawler)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\hp\bb-team-info-by-ingredient\.venv\Lib\site-packages\scrapy\middleware.py", line 88, in _from_settings
    mw = build_from_crawler(mwcls, crawler)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\hp\bb-team-info-by-ingredient\.venv\Lib\site-packages\scrapy\utils\misc.py", line 201, in build_from_crawler
    instance = objcls(*args, **kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\hp\bb-team-info-by-ingredient\food\food\pipelines.py", line 153, in __init__
    cell.font = Font(bold=True)
                ^^^^
NameError: name 'Font' is not defined
2025-07-14 11:26:20 [food.spiders.ingredients] WARNING: Crawling completed in 0:01:01.221685
2025-07-14 13:23:42 [food.spiders.ingredients] WARNING: Crawling completed in 0:01:00.595987
2025-07-14 13:31:49 [food.spiders.ingredients] WARNING: Crawling completed in 0:01:05.210250
2025-07-14 19:36:16 [food.spiders.ingredients] WARNING: Crawling completed in 0:02:38.838538
2025-07-14 19:44:17 [scrapy.core.engine] ERROR: Scraper close failure
Traceback (most recent call last):
  File "C:\Users\lenovo\bb-team-info-by-ingredient\.venv\Lib\site-packages\twisted\internet\defer.py", line 1088, in _runCallbacks
    current.result = callback(  # type: ignore[misc]
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\lenovo\bb-team-info-by-ingredient\food\food\pipelines.py", line 202, in close_spider
    self.sheet.calculate_dimensions.group(start_letter, end_letter, outline_level=1, hidden=False)
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'Worksheet' object has no attribute 'calculate_dimensions'. Did you mean: 'calculate_dimension'?
2025-07-14 19:44:17 [food.spiders.ingredients] WARNING: Crawling completed in 0:04:38.776558
2025-07-14 19:50:16 [scrapy.core.engine] ERROR: Scraper close failure
Traceback (most recent call last):
  File "C:\Users\lenovo\bb-team-info-by-ingredient\.venv\Lib\site-packages\twisted\internet\defer.py", line 1088, in _runCallbacks
    current.result = callback(  # type: ignore[misc]
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\lenovo\bb-team-info-by-ingredient\food\food\pipelines.py", line 203, in close_spider
    self.sheet.calculate_dimensions.group(start_letter, end_letter, outline_level=1, hidden=False)
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'Worksheet' object has no attribute 'calculate_dimensions'. Did you mean: 'calculate_dimension'?
2025-07-14 19:50:16 [food.spiders.ingredients] WARNING: Crawling completed in 0:01:46.656078
2025-07-14 19:55:44 [food.spiders.ingredients] WARNING: Crawling completed in 0:01:11.648802
2025-07-14 19:58:58 [scrapy.core.engine] ERROR: Scraper close failure
Traceback (most recent call last):
  File "C:\Users\lenovo\bb-team-info-by-ingredient\.venv\Lib\site-packages\twisted\internet\defer.py", line 1088, in _runCallbacks
    current.result = callback(  # type: ignore[misc]
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\lenovo\bb-team-info-by-ingredient\food\food\pipelines.py", line 205, in close_spider
    self.wb.save("foods.xlsx")
  File "C:\Users\lenovo\bb-team-info-by-ingredient\.venv\Lib\site-packages\openpyxl\workbook\workbook.py", line 386, in save
    save_workbook(self, filename)
  File "C:\Users\lenovo\bb-team-info-by-ingredient\.venv\Lib\site-packages\openpyxl\writer\excel.py", line 291, in save_workbook
    archive = ZipFile(filename, 'w', ZIP_DEFLATED, allowZip64=True)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\lenovo\AppData\Local\Programs\Python\Python312\Lib\zipfile\__init__.py", line 1336, in __init__
    self.fp = io.open(file, filemode)
              ^^^^^^^^^^^^^^^^^^^^^^^
PermissionError: [Errno 13] Permission denied: 'foods.xlsx'
2025-07-14 19:58:58 [food.spiders.ingredients] WARNING: Crawling completed in 0:01:01.750296
2025-07-14 20:00:43 [food.spiders.ingredients] WARNING: Crawling completed in 0:01:03.587270
